Source/WebCore/page/ScreenOrientationType.h

```diff
@@ -46,6 +46,16 @@ constexpr bool isLandscape(ScreenOrientationType type)
     return type == ScreenOrientationType::LandscapePrimary || type == ScreenOrientationType::LandscapeSecondary;
 }
 
+constexpr ScreenOrientationType naturalScreenOrientationType()
+{
+#if PLATFORM(IOS_FAMILY)
+    return ScreenOrientationType::PortraitPrimary;
+#else
+    // On Desktop, the natural orientation must be landscape-primary.
+    return ScreenOrientationType::LandscapePrimary;
+#endif
```

Comment:
For Apple platforms there are the following: iPhone, iPad, watchOS, tvOS, macOS. To me, it is obvious that iPhone is clearly portrait-primary. Not as clear that iPad, watchOS, and tvOS all should be.

Reply:
We had this discussion a while back with Tim and Wenson. iPad apparently doesn't have a clear "natural" orientation so we had decided to stick with PortraitPrimary like iPhone.

For watchOS and tvOS, we didn't explicitly discuss it so I had lumped them all together. I don't think the screen orientation API is super relevant to these platforms. We may want to refine this at some point though.

For now at least, this patch is not changing behavior for these platforms. The behavior change is for macOS at this point, where we have a clear regression.

Reply:
And presumably also changes behavior for all non-Apple platforms?

Reply:
If you wrote `PLATFORM(IOS)` it would be iPhone and iPad; that’s what I would do.

Reply:
> If you wrote `PLATFORM(IOS)` it would be iPhone and iPad; that’s what I would do.

But why take the risk to change the behavior on Watch and TV on the branch at this point?

Reply:
> But why take the risk to change the behavior on Watch and TV on the branch at this point?

OK.

Reply:
I don't mind changing behavior in a follow-up if people think TV and Watch should default to landscape (For TV I could see it make sense, for watch unsure). However, in this patch I'd rather focus on the macOS regression.

Reply:
Tv’s natural orientation should be landscape, if it even supports this API, but it’s not very important, not enough to change on a branch, certainly (but maybe on trunk).

For iPad, it seems the natural orientation is increasingly evolving to be landscape, but if it already was portrait, it doesn’t seem important to change it late in a release. (Maybe for a later release).

Reply:
I will follow-up for TV and iPad.

Reply:
Follow-up: https://github.com/WebKit/WebKit/pull/10803